xdsl/transforms/stencil_shape_minimize.py

```diff
@@ -91,6 +92,7 @@ def apply(self, ctx: MLContext, op: builtin.ModuleOp) -> None:
                     ]
                 )
             ).rewrite_module(op)
+        ShapeInferencePass().apply(ctx, op)
```

Comment:
I would recommend against this

Reply:
@n-io, what's the motivation for doing this inside the pass and not as part of the pass pipeline?

Reply:
The most immediate purpose is to undo the `InvalidateTemps()` rewrite pattern. The pass in general is for testing purposes, to allow restricting the stencil grid size to something small enough that for instance could run in a simulator. The idea is simple: invalidate temps, modify size on store ops, un-invalidate temps. This needs to happen before `stencil-shape-minimize` is run in order for that to work. The only alternative would be to separate this out from shape-minimize altogether, though running `stencil-shape-minimize{restrict=30,30,30}` seems to make some structural sense in my head instead of having it as a fully independent pass.

Reply:
Ah OK, that makes sense. The main thing that triggered me was calling a pass inside of another pass, for the same reason, as the contract is that a pass leaves the IR in a valid state, if the IR is in a valid state to begin with. In that sense it would make sense to split out the body of the shape inference pass into a helper method, and to call into that from here.

Reply:
I've seen this done in a few places, though was never sure why helper methods are preferred, including by `post_walk_func`?

Reply:
My understanding is that it's preferable to expose the least restrictive API when possible. For example, there's no need to pass the context when performing shape inference. It's probably also to make things more flexible in general, like you may want to only perform shape inference on a subset of your module.

Reply:
I'm not sure I follow - do I need to make some updates here?

Reply:
Yes, please! Could pull out the body of the shape inference pass into a python function that takes any operation, and no MLContext, and call that helper function from here instead of the pass?